pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +380,43 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
 		}
 
+		var networkPatchNeeded bool
+		// IF the VMI has a pod network, Check and patch if IP on VMI is not same as the one on Pod IP
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.NetworkSource.Pod == nil {
+				continue
+			}
+			for i, iface := range vmiCopy.Status.Interfaces {
+				if iface.Name == network.Name {
+					vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
```

Comment:
Wouldn't this patch every time? Shouldn't you patch only if the ip on the interface is differs from the one on the pod?

Reply:
Ah yes, good catch, changed now.